cloud/digitalocean/droplet/resources/droplet.go

```diff
@@ -171,17 +168,21 @@ func (r *Droplet) Apply(actual, expected cloud.Resource, immutable *cluster.Clus
 				found = true
 			} else {
 				logger.Info("Setting up VPN on Droplets... this could take a little bit longer...")
-				pubPath := local.Expand(immutable.ProviderConfig().SSH.PublicKeyPath)
-				privPath := strings.Replace(pubPath, ".pub", "", 1)
-				scp := scp.NewSecureCopier(immutable.ProviderConfig().SSH.User, masterIPPublic, "22", privPath, agent)
-				masterVpnIP, err := scp.ReadBytes("/tmp/.ip")
+				client := ssh.NewClient(masterIPPublic, providerConfig.SSH.Port,
+					providerConfig.SSH.User, providerConfig.SSH.PublicKeyPath)
+				err = client.Connect()
+				if err != nil {
+					return nil, nil, fmt.Errorf("Unable to connect to SSH: %v", err)
+				}
+
+				masterVpnIP, err := scp.ReadBytes(client, "/tmp/.ip")
```

Comment:
Do we want to pass in the client every time we call `ReadBytes` or do we want to cache it in memory somewhere and then tell it which file to read from?

Reply:
I guess we can encapsulate it into a struct. I've updated the SCP package to try to fix this. Let me know is it any better.